software/firmware/bootloader.py

```diff
@@ -144,4 +159,11 @@ def main(self):
             europi.b1._handler_both(europi.b2, self.exit_to_menu)
             europi.b2._handler_both(europi.b1, self.exit_to_menu)
 
-            script_class().main()
+            try:
+                script_class().main()
+            except Exception:
+                # in case we have the USB cable connected, print the stack trace for debugging
+                # otherwise, just halt and show the error message
+                print(f"[ERR ] Failed to run script: {err}")
+                traceback.print_exc()
```

Comment:
[question] Does `traceback` exist in micropython? I don't see it in the docs, but I do see [sys.print_exception](https://docs.micropython.org/en/latest/library/sys.html?highlight=traceback#sys.print_exception)

Reply:
My mistake; apparently it doesn't exist. I'll fix that.